### ISSUE TITLE
FR: @accessible should elide accessors for `protected` and `private` members

### ISSUE THREAD

**Issue by virusdave** [zio/zio#5014]
See title.

Use case: My Service has a `protected def` which is used by default implementations of the "real" methods in the `Service`.  I want to be able to override or access it in particular Service implementations, but not expose it outside the module.

Alternatively, allow me to explicitly annotate with `@hidden` or something (although this is probably less natural).

**Comment by adamgfraser** [comment]
Makes sense to me. Any of our macro wizards want to take a crack at this?

**Comment by hearnadam** [comment]
@virusdave do you still want this? I can take a look if you want.

**Comment by hearnadam** [comment]
Not pursued. The annotations will be deprecated for Scala 3 as it's no longer possible IIRC.
